lib/AST/ASTContext.cpp

```diff
@@ -3131,6 +3131,32 @@ AnyFunctionType *AnyFunctionType::withExtInfo(ExtInfo info) const {
   llvm_unreachable("unhandled function type");
 }
 
+AnyFunctionType *AnyFunctionType::getUncurriedFunction() {
+  assert(getCurryLevel() > 0 && "nothing to uncurry");
+
+  auto innerFunction = getResult()->castTo<AnyFunctionType>();
+  SmallVector<TupleTypeElt, 4> params{getInput()->getDesugaredType()};
+
+  if (auto tuple = innerFunction->getInput()->getAs<TupleType>())
```

Comment:
This will look through the ParenType that should be here if the function was a unary function that just took a tuple-typed argument.